packages/expo/src/utils/parseConstants.ts

```diff
@@ -0,0 +1,83 @@
+/* eslint-disable eslint-comments/disable-enable-pair */
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+
+/* --- Types ----------------------------------------------------------------------------------- */
+
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type Any$Todo = any
+
+export enum BRANCH_ENV {
+  development = 'development',
+  staging = 'staging',
+  production = 'production',
+}
+
+export type BranchEnvironment = `${BRANCH_ENV}`
+
+export type BranchConfig = {
+  devBranches: string[]
+  stageBranches: string[]
+  prodBranches: string[]
+}
+
+/** --- parseCOnstants() ----------------------------------------------------------------------- */
+/** -i- Parse expo-constants to get relevant app info
+ ** @param Constants - Your version of expo-constants
+ ** @param branchConfig - Config list of all branches matching dev / stage / prod */
+export const parseConstants = (
+  Constants: Any$Todo,
+  branchConfig: BranchConfig
+) => {
+  // Manifests
+  const { manifest, manifest2, expoConfig, platform } = Constants
+
+  // Local Debugger IP
+  const debuggerHostUrl: string | undefined = manifest?.debuggerHost || manifest2?.extra?.expoGo?.debuggerHost // prettier-ignore
+  const localUrl: string | undefined = debuggerHostUrl?.split(':').shift() // prettier-ignore
+
+  // Names
+  const branchName: string | undefined = manifest?.releaseChannel || manifest2?.metadata?.branchName // prettier-ignore
```

Comment:
Q:

Is this now a thing we do?

Reply:
Using branches?

Reply:
If that's what you mean, I suppose the answer is yes, as classic updates are now no more:

https://docs.expo.dev/archive/classic-updates/introduction/

> Classic Updates was a service by Expo that delivered updates to end-users using Expo CLI's expo publish command. It was supported by Expo through August 2022, when it was replaced by [EAS Update](https://docs.expo.dev/eas-update).

https://docs.expo.dev/versions/latest/sdk/updates/#updatesreleasechannel

I do read that `releaseChannel` will always be `'default'` from now on though, so I'll switch the order of these I think (or remove `manifest?.releaseChannel` entirely)